packages/next/src/shared/lib/xss-protection.ts

```diff
@@ -0,0 +1,46 @@
+export type UntrustedHref = string
+export interface TrustedHref {
+  unsafeHref: {
+    __href: string
+  }
+}
+
+// TODO: Should this handle special chars?
```

Comment:
Doesn't seem like it: https://vercel.slack.com/archives/C01RUTYJYHW/p1713535684819809?thread_ts=1712932117.992689&cid=C01RUTYJYHW